smartBugs.py

```diff
@@ -12,58 +12,45 @@
 from src.execution.execution_task import Execution_Task
 from src.execution.execution_configuration import Execution_Configuration
 from src.logger import logs
-from src.interface.cli import get_config, get_remote_dataset, is_remote_dataset
-
-
-cfg_dataset_path = os.path.join(os.path.dirname(__file__), 'config', 'dataset', 'dataset.yaml')
-with open(cfg_dataset_path, 'r') as ymlfile:
-    try:
-        cfg_dataset = yaml.safe_load(ymlfile)
-    except yaml.YAMLError as exc:
-        logs.print(exc)
-
+from src.interface.cli import get_config, cfg_dataset
+from src.utils import COLSTATUS,COLRESET,COLERR
 
 def create_tasks(conf: 'Execution_Configuration') -> List['Execution_Task']:
     files_to_analyze = []
 
     if conf.datasets:
         for dataset in conf.datasets:
             base_name = dataset.split('/')[0]
-            if is_remote_dataset(cfg_dataset, base_name):
-                remote_info = get_remote_dataset(cfg_dataset, base_name)
-                base_path = remote_info['local_dir']
+            dataset_info = cfg_dataset[base_name]
+            if isinstance(dataset_info,dict) and 'url' in dataset_info and 'local_dir' in dataset_info: 
```

Comment:
I would prefer to have this check inside a method named `is_remote_dataset` because it's more obvious when reading the code.

Reply:
I think the options are to encapsulate the logic completely, like `is_remote_info/get_remote_info` or to let the tests and assignments access the  data structure directly, as I suggested. Mixing it like
```
if is_remote_info(dataset_info):
       ... dataset_info['url'] ... dataset_info['local_dir'] ...
```
has the disadvantage of obfuscating correctness. When checking the conditions for accessing `dataset_info` as a dict with components `url` and `local_dir` explicitly, before accessing it, correctness is obvious.

Encapsulation is fine, but I found the name `get_remote_dataset` misleading, better use `get_remote_info` or something similar. In fact, I checked twice to make sure that I hadn't overlooked the instructions where the dataset is actually loaded. So we could have something like
```
if is_remote_inf(dataset_info):
    (url,local_dir) = get_remote_info(dataset_info)
```
I modified the code accordingly.

A really clean solution would hide the structure of the configuration, and of the dataset file, completely. But this is maybe too much of an overhead for the moment.